cli/azd/pkg/project/remote_build_helper.go

```diff
@@ -0,0 +1,115 @@
+package project
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerregistry/armcontainerregistry"
+	"github.com/azure/azure-dev/cli/azd/pkg/containerregistry"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/benbjohnson/clock"
+)
+
+type RemoteBuildHelper struct {
+	env                *environment.Environment
+	remoteBuildManager *containerregistry.RemoteBuildManager
+	console            input.Console
+	clock              clock.Clock
+}
+
+func NewRemoteBuildHelper(
+	env *environment.Environment,
+	remoteBuildManager *containerregistry.RemoteBuildManager,
+	console input.Console,
+	clock clock.Clock,
+) *RemoteBuildHelper {
+	return &RemoteBuildHelper{
+		env:                env,
+		remoteBuildManager: remoteBuildManager,
+		console:            console,
+		clock:              clock,
+	}
+}
+
+func (rh *RemoteBuildHelper) RunRemoteBuild(
+	ctx context.Context, serviceConfig *ServiceConfig, target *environment.TargetResource, updateProgress func(string),
+) error {
+	dockerOptions := getDockerOptionsWithDefaults(serviceConfig.Docker)
+
+	if !filepath.IsAbs(dockerOptions.Path) {
+		dockerOptions.Path = filepath.Join(serviceConfig.Path(), dockerOptions.Path)
+	}
+
+	if !filepath.IsAbs(dockerOptions.Context) {
+		dockerOptions.Context = filepath.Join(serviceConfig.Path(), dockerOptions.Context)
+	}
+
+	if dockerOptions.Platform != "linux/amd64" {
+		return fmt.Errorf("remote build only supports linux/amd64 platform")
+	}
+
+	updateProgress("Packing remote build context")
+
+	contextPath, dockerPath, err := containerregistry.PackRemoteBuildSource(ctx, dockerOptions.Context, dockerOptions.Path)
+	if contextPath != "" {
+		defer os.Remove(contextPath)
+	}
+	if err != nil {
+		return err
+	}
+
+	updateProgress("Uploading remote build context")
+
+	registryName, err := RegistryName(ctx, rh.env, serviceConfig)
+	if err != nil {
+		return err
+	}
+
+	registryResourceName := strings.Split(registryName, ".")[0]
+
+	source, err := rh.remoteBuildManager.UploadBuildSource(
+		ctx, target.SubscriptionId(), target.ResourceGroupName(), registryResourceName, contextPath)
+	if err != nil {
+		return err
+	}
+
+	imageName := fmt.Sprintf("%s/%s-%s:azd-deploy-%d",
+		strings.ToLower(serviceConfig.Project.Name),
+		strings.ToLower(serviceConfig.Name),
+		strings.ToLower(rh.env.Name()),
+		rh.clock.Now().Unix())
```

Comment:
Instead of manually constructing the image name here we should look at reusing the `containerHelper.RemoteImageTag(...)` that does all the processing to generate the correct image name based on the service configuration, etc.